master/buildbot/statistics/stats_service.py

```diff
@@ -0,0 +1,67 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from twisted.internet import defer
+from twisted.internet import threads
+from twisted.python import log
+
+from buildbot.util import service
+from buildbot.statistics.storage_backends import StatsStorageBase
+
+
+class StatsService(service.ReconfigurableServiceMixin, service.AsyncMultiService):
+    """
+    A middleware for passing on statistics data to all storage backends.
+    """
+    def __init__(self, master):
+        service.AsyncMultiService.__init__(self)
+        self.setName('StatsService')
+        log.msg("Creating StatsService")
+        self.master = master
+        self.registeredStorageServices = []
+
+    def reconfigServiceWithBuildbotConfig(self, new_config):
```

Comment:
please look at the new reporters services to see how buildbotService can be made